docs/section5/Rule5-39.md

```diff
@@ -1,32 +1,35 @@
-
 # Envelope - Rule 5-39  
-
+**Schema Version** 0.0.29
+**Primary Rule:** False  
 **Rule ID:** 5-39  
-**Rule Description:** It is acceptable to use either an annual average ground temperature or monthly average ground temperatures for calculation of heat loss through basement floors.  
-**Rule Assertion:** B-RMR ASHRAE229:weather.monthly_ground_temperature != NULL  
+**Rule Description:** It is acceptable to use either an annual average ground temperature or monthly average ground temperatures for calculation of heat loss through basement floors.
 **Appendix G Section:** Section 5 Envelope  
 **Appendix G Section Reference:** Section G3.1-14(b) Building Envelope Modeling Requirements for the Proposed design and Baseline  
 
-**Applicability:** All required data elements exist for B_RMR  
-**Applicability Checks:**  None  
-
-**Manual Check:** Yes  
-**Evaluation Context:** Each Data Element  
+**Applicability:** All required data elements exist for B_RMD
+**Applicability Checks:**  
+  1. B_RMD contains at least 1 floor surface adjacent to ground
+ 
+**Evaluation Context:** RMD 
 **Data Lookup:** None  
 **Function Call:** None  
 
 ## Rule Logic:  
+**Applicability Check 1**
+- If for-loop completes and no floor surfaces were found adjacent to ground, rule is not applicable: `is_applicable = false`
+- For each surface in the B_RMD: `for surface in B_RMD...Surface:` 
+  - If the surface is a floor and adjacent to ground, rule is applicable and continue to rule assertion: `if surface.classification == FLOOR and surface.adjacent_to == GROUND: is_applicable = true; break`
 
-**Rule Assertion:**  
-
-- Case 1: If B-RMR includes annual average or monthly average ground temperature: `if B_RMR.ASHRAE229.weather.monthly_ground_temperature: PASS`
-
-- Case 2: Else: `else: FAIL`
+**Rule Assertion:**
+- Case 1: If B_RMD has no floor surfaces adjacent to ground: `if not is_applicable: NOT_APPLICABLE`
+- Case 2: If RPD includes a ground temperature schedule: `if RPD.weather.ground_temperature_schedule: UNDETERMINED`
```

Comment:
One quick observation is - do we want to add data association verification here?
My thinking is we may elevate the association to a schema validation level so before we reaching this rule checking logic, the RCT already checked whether the `ground_temperature_schedule` is actually in the `Schedule` data group.

Reply:
Is the ruleset still evaluated if a json does not pass the schema validation level?

I believe we would still want the schema to be considered valid, but only this rule would result in UNDETERMINED outcome.

Reply:
I see, so that the schema would be considered invalid if the `id` for `weather.ground_temperature_schedule` does not exist in the `Schedule` data group. That makes sense to me.

Reply:
Do you need me to make any changes to the RDS to specify this?

Reply:
I just wondering if it make sense for RDS to perform such check vs. RCT's validation module checks it. The current validation module performs schema and non-schema validation functions. The non-schema validation currently include only unique ID check. We can add the associations in the non-schema validation function too. 
@claperle @JacksonJ-KC @mkarpman what do you think?

Reply:
Discussed on the RDS call, 12/14, this validation makes sense for the RCT level. However, many rule tests would then become invalid so there also needs to be a way for rule tests to bypass this validation

Reply:
We just added a way to bypass the association checks in rule tests.